src/components/progress/progress-bar.js

```diff
@@ -80,13 +80,14 @@ export const BProgressBar = /*#__PURE__*/ Vue.extend({
     },
     computedMax() {
       // Prefer our max over parent setting
-      const max = toFloat(this.max)
-      return isNaN(max) ? toFloat(this.bvProgress.max, 100) : max
+      // Default to `100` for invalid values (`-x`, `0`, `NaN`)
+      const max = toFloat(this.max) || toFloat(this.bvProgress.max, 0)
+      return max > 0 ? max : 100
     },
     computedPrecision() {
       // Prefer our precision over parent setting
-      const precision = toInteger(this.precision)
-      return isNaN(precision) ? toInteger(this.bvProgress.precision, 0) : precision
+      // Default to `0` for invalid values (`-x`, `NaN`)
+      return Math.max(toInteger(this.precision) || toInteger(this.bvProgress.precision, 0), 0)
```

Comment:
If the local precision is 0, this will chose the parent precision.... which might not be what is desired... has to be if no local precision is provided then fall back to parent precision.

Reply:
```js
return Math.max(toInteger(this.precision, toInteger(this.bvProgress.precision, 0)), 0)